codebase/codebase.go

```diff
@@ -0,0 +1,68 @@
+package codebase
+
+import "errors"
+
+// Codebase defines all parameters those are useful to associate Che Editor's window to a WI
+type Codebase struct {
+	Repository string `json:"repository"`
+	Branch     string `json:"branch"`
+	FileName   string `json:"filename"`
+	LineNumber int    `json:"linenumber"`
+}
+
+// Following keys define attribute names in the map of Codebase
+const (
+	RepositoryKey = "repository"
+	BranchKey     = "branch"
+	FileNameKey   = "filename"
+	LineNumberKey = "linenumber"
+)
+
+// ToMap converts Codebase to a map of string->Interface{}
+func (c *Codebase) ToMap() map[string]interface{} {
```

Comment:
@sbose78 yes we can. But  I would not like to use `codebase` in the method names because receiver describes that.
And I was trying to align to [rendering stuff](https://github.com/almighty/almighty-core/blob/master/rendering/markup_content.go#L16).